src/astarte-client/client.ts

```diff
@@ -316,6 +318,38 @@ class AstarteClient {
     return fromAstarteDeviceDTO(response.data);
   }
 
+  async insertDeviceAlias(deviceId: AstarteDevice['id'], aliasKey: string, aliasValue: string): Promise<unknown> {
+    const response = await this.$patch(
+      this.apiConfig.deviceInfo({ deviceId, ...this.config }),
+      { aliases: {[aliasKey]: aliasValue}},
+    );
+    return response;
+  }
+
+  async deleteDeviceAlias(deviceId: AstarteDevice['id'], aliasKey: string): Promise<unknown> {
+    const response = await this.$patch(
+      this.apiConfig.deviceInfo({ deviceId, ...this.config }),
+      { aliases: {[aliasKey]: null}},
+    );
+    return response;
+  }
+
+  async insertDeviceMetadata(deviceId: AstarteDevice['id'], metadataKey: string, metadataValue: string): Promise<unknown> {
+    const response = await this.$patch(
+      this.apiConfig.deviceInfo({ deviceId, ...this.config }),
+      { metadata: {[metadataKey]: metadataValue}},
+    );
+    return response;
+  }
+
+  async deleteDeviceMetadata(deviceId: AstarteDevice['id'], metadataKey: string): Promise<unknown> {
+    const response = await this.$patch(
+      this.apiConfig.deviceInfo({ deviceId, ...this.config }),
+      { metadata: {[metadataKey]: null}},
+    );
+    return response;
+  }
```

Comment:
I think it would be better to use a Params object, like in the other APIs, instead of multiple single params.
Also I would define a proper type for the return value, if possible. Maybe:
- avoid returning values, if not used: => Promise<void>
- or return a suitable object derived from `response.data`